x/oracle/types/keys.go

```diff
@@ -110,3 +113,17 @@ func ReportsOfValidatorPrefixKey(reqID RequestID, val sdk.ValAddress) []byte {
 func ReportersOfValidatorPrefixKey(val sdk.ValAddress) []byte {
 	return append(ReporterStoreKeyPrefix, val.Bytes()...)
 }
+
+// GetEscrowAddress returns the escrow address for the specified channel and request key.
+// The escrow address follows the format as outlined in ADR 028:
+// https://github.com/cosmos/cosmos-sdk/blob/master/docs/architecture/adr-028-public-key-addresses.md
+func GetEscrowAddress(requestKey, portID, channelID string) sdk.AccAddress {
+	contents := fmt.Sprintf("%s/%s/%s", requestKey, portID, channelID)
+
+	// ADR 028 AddressHash construction
+	preImage := []byte(Version)
```

Comment:
Should we use `func Hash(typ string, key []byte) []byte` instead where `typ` is `Version`?

Reply:
Oh, the function will be available in the future.

https://github.com/cosmos/cosmos-sdk/pull/8415

May use the current for now.